.github/workflows/ci.yml

```diff
@@ -148,32 +148,8 @@ jobs :
           java-version : '11'
           check-latest : true
 
-      - name : Publish to Maven Local
-        run : ./gradlew clean publishToMavenLocal --no-build-cache --no-daemon --stacktrace --no-parallel
-
-  publish-snapshot :
-    runs-on : ubuntu-latest
-    if : github.repository == 'square/anvil' && github.ref == 'refs/heads/main'
```

Comment:
Looking around to see what other options we have since workflows can't reference each other and it seems like we lose some of the existing checks by moving this out of the main ci workflow; what do you think about us just modifying this branch condition as needed when making temporary snapshots? If I'm understanding correctly, it looks like we'd need to do something like that with the new workflow file anyway since it's still gating on `main`

Reply:
The new workflow does not need to be edited in order to publish other branches.  Here's the `on` block:

```yaml

 on:
   workflow_dispatch:
   push:
     branches:
       - main
```

With that logic, it will run automatically any time there's a push to `main`.  But the `workflow_dispatch` node means that you can trigger it manually from the `Actions` tab, then select the branch you want to target.

It winds up looking like this:

<img width="1316" alt="image" src="https://github.com/square/anvil/assets/9890900/27568b57-884c-4444-ae17-c5fa2848d154">

Reply:
Oh nice, thanks for explaining! One last question/concern is differentiating the snapshots. My understanding right now is that running the workflow on a PR would clobber the existing snapshot generated from main unless someone first manually updated the version in gradle.properties which could cause issues if someone was already relying on the existing snapshot. Is there a way we can guarantee the PR-initiated workflow would use a custom version name?

Reply:
You could do something like this https://github.com/freeletics/khonshu/blob/main/.github/workflows/publish-snapshot.yml#L36-L39 that would would publish the snapshot as `<branch-name>-SNAPSHOT` if the workflow doesn't run on main. The env var takes precedence over `VERSION_NAME` in `gradle.properties`.

Reply:
:+1: that setup looks awesome, thanks for sharing!

Reply:
Discussed more with Rick offline, for now folks using [pinned versions](https://medium.com/@vanniktech/pinning-a-snapshot-dependency-to-a-specific-version-b477f79469fc) is sufficient for this concern. Maybe we'll look at updating the workflow logic later as well but for now this should be good